packages/runtime/container-runtime/src/test/gc/gc.perf.spec.ts

```diff
@@ -0,0 +1,227 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { ICriticalContainerError } from "@fluidframework/container-definitions";
+import { IGarbageCollectionData } from "@fluidframework/runtime-definitions/internal";
+import {
+	MockLogger,
+	createChildLogger,
+	mixinMonitoringContext,
+} from "@fluidframework/telemetry-utils/internal";
+import { BenchmarkType, benchmark } from "@fluid-tools/benchmark";
+
+import {
+	GCNodeType,
+	GarbageCollector,
+	IGCMetadata,
+	IGarbageCollectionRuntime,
+	IGarbageCollectionSnapshotData,
+	IGarbageCollectionState,
+	IGarbageCollector,
+	IGarbageCollectorCreateParams,
+	stableGCVersion,
+} from "../../gc/index.js";
+import { ContainerRuntimeGCMessage } from "../../messageTypes.js";
+import { pkgVersion } from "../../packageVersion.js";
+import { createTestConfigProvider } from "./gcUnitTestHelpers.js";
+
+type GcWithPrivates = IGarbageCollector & {
+	baseSnapshotDataP: Promise<IGarbageCollectionSnapshotData | undefined>;
+	initializeOrUpdateGCState: () => Promise<void>;
+};
+
+function createGarbageCollector(params: {
+	deleteSweepReadyNodes: (sweepReadyRoutes: string[]) => string[];
+	getGCData: () => Promise<IGarbageCollectionData>;
+	logger: MockLogger;
+	createParams?: Partial<IGarbageCollectorCreateParams>;
+	gcBlobsMap?: Map<string, any>;
+	gcMetadata?: IGCMetadata;
+	closeFn?: (error?: ICriticalContainerError) => void;
+	isSummarizerClient?: boolean;
+}): GcWithPrivates {
+	const {
+		createParams = {},
+		gcBlobsMap = new Map(),
+		gcMetadata = {},
+		closeFn = () => {},
+		isSummarizerClient = true,
+		getGCData,
+		deleteSweepReadyNodes,
+		logger,
+	} = params;
+	const testPkgPath = ["testPkg"];
+
+	const getNodeType = (nodePath: string) => {
+		if (nodePath.split("/").length !== 2) {
+			return GCNodeType.Other;
+		}
+		return GCNodeType.DataStore;
+	};
+
+	// The runtime to be passed to the garbage collector.
+	const gcRuntime: IGarbageCollectionRuntime = {
+		updateStateBeforeGC: async () => {},
+		getGCData,
+		updateUsedRoutes: (usedRoutes: string[]) => {
+			return { totalNodeCount: 0, unusedNodeCount: 0 };
+		},
+		deleteSweepReadyNodes,
+		updateTombstonedRoutes: (tombstoneRoutes: string[]) => {},
+		getNodeType,
+		getCurrentReferenceTimestampMs: () => Date.now(),
+		closeFn,
+	};
+
+	let metadata = createParams.metadata;
+	const existing = createParams.baseSnapshot !== undefined;
+	// For existing, add container runtime metadata which is required for GC to be enabled.
+	if (existing) {
+		metadata = {
+			...metadata,
+			...gcMetadata,
+			gcFeature: gcMetadata.gcFeature ?? stableGCVersion,
+			summaryFormatVersion: 1,
+			message: undefined,
+		};
+	}
+
+	return GarbageCollector.create({
+		...createParams,
+		runtime: gcRuntime,
+		gcOptions: createParams.gcOptions ?? {},
```

Comment:
nit: several things are referencing createParams, they can just be omitted since createParams is spread at the top.

For this one in particular, you could do `gcOptions: {}` before `...createParams`.

I'm assuming you just copied this from some other test, so up to you what to do.

Reply:
Cleaned it all up :-)